modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/SleepTimer.kt

```diff
@@ -43,6 +56,23 @@ class SleepTimer @Inject constructor(@ApplicationContext private val context: Co
         }
         createAlarm(time.timeInMillis)
     }
+    suspend fun restartSleepTimerIfApplies(currentEpisodeUuid: String, isSleepTimerRunning: Boolean, onRestartSleepAfterTime: () -> Unit, onRestartSleepOnEpisodeEnd: suspend () -> Unit) {
+        val diffTimeInMillis = System.currentTimeMillis() - lastTimeHasFinishedInMillis
+        val diffInMinutes = diffTimeInMillis / (1000 * 60) // Convert to minutes
+
+        if (diffInMinutes < MIN_TIME_TO_RESTART_SLEEP_TIMER_IN_MINUTES && !lastEpisodeUuidAutomaticEnded.isNullOrEmpty() && currentEpisodeUuid != lastEpisodeUuidAutomaticEnded) {
+            onRestartSleepOnEpisodeEnd()
+            analyticsTracker.track(PLAYER_SLEEP_TIMER_RESTARTED, mapOf(TIME_KEY to END_OF_EPISODE_VALUE))
+        } else if (diffInMinutes < MIN_TIME_TO_RESTART_SLEEP_TIMER_IN_MINUTES && lastSleepAfterTimeInMinutes != 0 && !isSleepTimerRunning) {
+            analyticsTracker.track(PLAYER_SLEEP_TIMER_RESTARTED, mapOf(TIME_KEY to lastSleepAfterTimeInMinutes * 60)) // Convert to seconds
+            sleepAfter(lastSleepAfterTimeInMinutes, onRestartSleepAfterTime)
+        }
+    }
+    fun sleepTimerWasTriggeredForEndOfEpisode(uuid: String?) {
```

Comment:
It's a bit hard for me to read the name of this function clearly. Could you rename it something simpler? Even `setEndOfEpisodeUuid()` would be enough I believe.

Also, this function shouldn't accent nullable uuid.